src/topology/metric_space/closeds.lean

```diff
@@ -211,12 +211,14 @@ instance closeds.compact_space [compact_space α] : compact_space (closeds α) :
   let F := {f : closeds α | (f : set α) ⊆ s},
   refine ⟨F, _, λ u _, _⟩,
   -- `F` is finite
-  { apply @finite_of_finite_image _ _ F coe,
-    { exact set_like.coe_injective.inj_on F },
+  { apply @finite.of_finite_image _ _ F coe,
     { refine fs.finite_subsets.subset (λb, _),
+      apply_instance,
```

Comment:
Does replacing the apply with a refine make this go away?

Reply:
If you mean replace the `refine` with `apply`, then yes it did.